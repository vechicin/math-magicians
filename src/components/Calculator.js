import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [operand, setOperand] = useState({});

  const printOn = (event) => {
    const buttonName = event.target.textContent;
    setOperand(calculate(operand, buttonName));
  };

  return (
    <div className="container">
      <h1 className="calculator-title">Math Magicians</h1>
      <h2 className="calculator-subtitle">Let&apos;s do some math!</h2>
      <div className="calculator-grid">
        <div className="output">
          {operand.total}
        &nbsp;
          {operand.operation}
        &nbsp;
          {operand.next}
        </div>
        <button type="button" onClick={printOn}>AC</button>
        <button type="button" onClick={printOn}>+/-</button>
        <button type="button" onClick={printOn}>%</button>
        <button type="button" className="operator" onClick={printOn}>รท</button>
        <button type="button" onClick={printOn}>7</button>
        <button type="button" onClick={printOn}>8</button>
        <button type="button" onClick={printOn}>9</button>
        <button type="button" className="operator" onClick={printOn}>x</button>
        <button type="button" onClick={printOn}>4</button>
        <button type="button" onClick={printOn}>5</button>
        <button type="button" onClick={printOn}>6</button>
        <button type="button" className="operator" onClick={printOn}>-</button>
        <button type="button" onClick={printOn}>1</button>
        <button type="button" onClick={printOn}>2</button>
        <button type="button" onClick={printOn}>3</button>
        <button type="button" className="operator" onClick={printOn}>+</button>
        <button type="button" className="occupy-two" onClick={printOn}>0</button>
        <button type="button" onClick={printOn}>.</button>
        <button type="button" className="operator" onClick={printOn}>=</button>
      </div>
    </div>
  );
};

export default Calculator;
