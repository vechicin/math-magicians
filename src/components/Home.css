* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

body {
  color: #fff;
  display: flex;
  flex-direction: row;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

ul {
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: flex-end;
  gap: 5%;
  list-style: none;
  padding: 2%;
}

h1 {
  margin: 5%;
}

h2 {
  margin: 5%;
}

p {
  margin: 5%;
}
