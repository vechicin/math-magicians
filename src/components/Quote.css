#root {
  width: 100%;
}

.quote-title {
  margin: 0%;
  padding: 5%;
}

.quote {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 30px;
  padding: 10%;
}
