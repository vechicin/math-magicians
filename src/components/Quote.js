import React from 'react';
import Navbar from './Navbar';
import './Quote.css';

const Quote = () => (
  <div>
    <Navbar />
    <h1 className="quote-title">Math Magicians</h1>
    <div className="quote">
      Mathematics is not about numbers, equations, computations,
      or algorithms: it is about understanding. -William Paul Thurston
    </div>
  </div>
);

export default Quote;
