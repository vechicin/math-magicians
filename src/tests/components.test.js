import React from "react";
import { render, screen } from '@testing-library/react';
import Calculator from "../components/Calculator";
import Home from "../components/Home";
import Navbar from "../components/Navbar";
import Quote from "../components/Quote";

describe("test components", () => {
  test("Calculator tests", () => {
    render(<Calculator />)
    screen.getByText('AC')
  });
  it("Home tests", () => {
    render(<Home />)
    screen.getByText('Welcome')
  })
});
