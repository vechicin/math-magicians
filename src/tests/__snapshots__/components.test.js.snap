// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components Calculator tests 1`] = `
<div
  className="calculator-container"
>
  <div
    className="title-container"
  >
    <h1
      className="calculator-title"
    >
      Math Magicians
    </h1>
    <h2
      className="calculator-subtitle"
    >
      Let's do some math!
    </h2>
  </div>
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
       
       
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="occupy-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`test components Navbar tests 1`] = `
<div>
  <h1
    className="quote-title"
  >
    Math Magicians
  </h1>
  <div
    className="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
  </div>
</div>
`;
